lib/modules/manager/kustomize/extract.ts

```diff
@@ -50,10 +51,17 @@ export function extractResource(base: string): PackageDependency | null {
     return null;
   }
 
-  const { path } = match.groups;
+  const { path, queryString } = match.groups;
+  const params = querystring.parse(queryString);
+  const refParam = Array.isArray(params.ref) ? params.ref[0] : params.ref;
+  const versionParam = Array.isArray(params.version)
+    ? params.version[0]
+    : params.version;
+  const currentValue = refParam ?? versionParam;
```

Comment:
Kustomize docs allow usage of `version` query parameter as well (`ref` trumps it, though). So added it as a fallback in case `ref` parameter is missing. Since it's not related to the actual fix, let me know if you'd prefer not to support `version` parameter (for the time being).

Reply:
What's the behavior now vs before if there's a query string present but it contains neither ref or version params?